docs/docs/themes/shadowing.md

```diff
@@ -0,0 +1,215 @@
+  </Card>
+)
+```
+
+This workflow isn’t too bad, especially since the component is relatively straightforward. However, it could be optimized in scenarios where you want to wrap a component or pass a different prop without having to worry about the component’s internals.
+
+## Importing the Shadowed Component
+
+In the above example it might be preferable to be able to import the `Bio` component and wrap it with your `Card`. With extending you can do the following instead:
+
+```js:title=src/gatsby-theme-blog/components/bio.js
+import React from "react"
+import { Author } from "gatsby-theme-blog/src/components/bio"
+import Card from "../components/card"
+
+export default props => (
+  <Card>
+    <Author {...props} />
+  </Card>
+)
+```
+
+This is a quick and efficient way to customize rendering without needing to worry about the implementation details of the component you’re looking to customize. Extending the shadowed component means you can use composition, leveraging a great feature from React.
+
+### Applying New Props
+
+In some cases there could be a component with different variants without an API to modify it outside of shadowing. With component extending you can import that component and then add your new prop to change it.
+
+If `NewsletterButton` accepts a `variant` prop which changes the look and colors of the button, you can use it when you extend the component. Below, `NewsletterButton` is re-exported and `variant="link"` is added in the shadowed file to override its default value.
+
+```js:title=src/gatsby-theme-blog/components/newsletter/button.js
+import { NewsletterButton } from "gatsby-theme-blog/src/components/newsletter"
+
+export default props => <NewsletterButton {...props} variant="link" />
+```
+
+## Using the CSS Prop
+
+In addition to passing a different prop to a component you’re extending, you might want to apply CSS using the [Emotion CSS prop](/docs/emotion/). This will allow you to change the styling of a particular component without changing any of its functionality.
+
+```js:title=src/gatsby-theme-blog/components/newsletter/button.js
+import { NewsletterButton } from "gatsby-theme-blog/src/components/newsletter"
+
+export default props => (
+  <NewsletterButton
+    css={{
+      backgroundColor: "rebeccapurple",
+      color: "white",
+      boxShadow: "none",
+    }}
+    {...props}
+  />
+)
+```
+
+**Note:** For this approach to work NewsletterButton has to accept a className property.
+
+```js:title=src/gatsby-plugin-theme-ui/index.js
+import theme from "gatsby-plugin-theme-ui/src"
+
+const { colors } = theme
+export default {
+  ...theme,
+  colors: {
+    ...colors,
+    primary: "tomato",
+  },
+}
+```
+
+This provides a nice interface to extend an object if you want to change a couple values from the defaults.
+
+## How Much Shadowing is too Much Shadowing?
+
+If you find yourself shadowing a large number of components in a particular theme it might make sense to fork and modify the theme instead.
```

Comment:
Maybe a mention of the `-core` pattern here?

```suggestion
If you find yourself shadowing a large number of components in a particular theme it might make sense to fork and modify the theme instead. The official themes support this pattern using a set of `-core` parent themes. `gatsby-theme-blog` relies on `gatsby-theme-blog-core` so you can fork `gatsby-theme-blog` (or skip it completely) to render your own components without having to worry about dealing with any of the page creation or data sourcing logic.
```